docs/modules/pipelines/pages/job-monitoring.adoc

```diff
@@ -0,0 +1,56 @@
+= Monitoring Jobs
+:description: To ensure the reliability of streaming jobs and their workloads, try implementing some basic monitoring. With Hazelcast, you can monitor the status of streaming jobs to quickly identify job failures for troubleshooting.
+
+{description}
+
+== Monitoring for Status Changes
+
+There are two ways to monitor for job status changes. You can either poll the `Job.getStatus()` method separately for each job or add a `JobStatusListener` to your jobs.
+
+== Using a JobStatusListener
+
+The `JobStatusListener` has several advantages over polling for the job status. 
+
+- You can write custom code for execution on status change. For example, triggering an alert or writing the changed status to a dashboard.
```

Comment:
This applies to `Job.getStatus()` as well. The difference is that the listener notifies you immediately on status change whereas in polling, you won't know that the status has changed until you request the latest status. In order to get status changes sooner, you will poll more frequently, which will decrease performance. `JobStatusListener` is introduced to optimize that kind of logic.

The sentence also implies that status changes are guaranteed to be caught, which is untrue. For example, when a job is submitted, it is queued for execution and the method returns immediately (without waiting for the RUNNING status). For operations that require the job to be running, the user must wait until the status change. Until the introduction of `JobStatusListener`, people used `Job.getStatus()` in a loop to achieve this. `JobStatusListener` unfortunately cannot be used as an efficient replacement for this case since the listener is attached to the job after it is submitted and the job may start running before the listener attachment, which will cause the RUNNING status to be missed by the listener. The user should be informed about this caveat and provided a code snippet for busy-wait.

Reply:
Hi @burakgok, Could you provide a code snippet for `busy-wait` so that I can make these changes all at once?